# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLImageElement.h
# Commit: a8f10a064b69
# Full Hash: a8f10a064b69c36a2949ea6fd8ffd26bad6fc2d1
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1694741
# File Overlap Count: 1
# Description:
#   Bug 1694741 - Part 4: Factor out the setter of mCurrentDensity and mResponsiveSelector into separate functions. r=emilio
#   
#   So we can set the density during UpdateResponsiveSource(), and these functions
#   make us easier to restyle the image when we need to update the responsive
#   source synchronously in the patch series.
# ==============================================================================

diff -r ce1f3f158029 -r a8f10a064b69 dom/html/HTMLImageElement.h
--- a/dom/html/HTMLImageElement.h	Mon Aug 22 20:18:38 2022 +0000
+++ b/dom/html/HTMLImageElement.h	Mon Aug 22 20:18:38 2022 +0000
@@ -331,9 +331,10 @@
   // create a ResponsiveSelector.
 
   // If the node's srcset/sizes make for an invalid selector, returns
-  // false. This does not guarantee the resulting selector matches an image,
+  // nullptr. This does not guarantee the resulting selector matches an image,
   // only that it is valid.
-  bool TryCreateResponsiveSelector(Element* aSourceElement);
+  already_AddRefed<ResponsiveImageSelector> TryCreateResponsiveSelector(
+      Element* aSourceElement);
 
   MOZ_CAN_RUN_SCRIPT CSSIntPoint GetXY();
   virtual JSObject* WrapNode(JSContext* aCx,
@@ -400,6 +401,9 @@
            GetParentElement()->IsHTMLElement(nsGkAtoms::picture);
   }
 
+  void SetResponsiveSelector(RefPtr<ResponsiveImageSelector>&& aSource);
+  void SetDensity(double aDensity);
+
   bool mInDocResponsiveContent;
 
   RefPtr<ImageLoadTask> mPendingImageLoadTask;
